doc/ax/ax.txt

```diff
@@ -0,0 +1,2095 @@
+Language. The design of AX uses concepts from these languages and refrains
+from deviating from their syntax too significantly. Specifically, it does not
+aim to change what are already well established and heavily used language
+syntaxes within the Visual Effects industry; on the contrary, AX aims to provide
+an interface to which a user with previous or little programming experience can
+easily transition to. However it does introduce new concepts where the the
+creators deemed necessary to provide the best and most representative syntax
+relating to the design for OpenVDB volume and point modification.
+
+
+@section axprograms AX Programs
+@par
+AX programs represent a set of statements which read and write to geometry,
+with each program designed in such a way that it can be executed in a highly
+parallelized framework to access and update individual geometric components.
+For example, consider a mesh with point, vertex and primitive
+attributes. AX programs are designed to run independently across "elements" of
+input geometry and process "attributes" from a particular element. In this case,
+the element level would correlate to either points, vertexes or primitives, and
+a single element would be a unique instance of one of these. These programs are
+at their most efficient writing to the currently processing element, however do
+provide varying levels of access to the geometry as a whole.
+@par
+Note that native AX only supports the modification of OpenVDB data. Currently,
+that means that native AX programs can be written and built for execution over
+OpenVDB Points or OpenVDB Volumes. For any given AX program, the execution context
+(what it's processing) may change what functionality is available and, more
+importantly, the behaviour of reading/writing from attributes. See the @ref axexecutionxcontext
+for details.
+@par
+For details of extending AX for custom geometry, see the @subpage openvdbaxapi
+developer documentation. Note that custom geometry support requires C++ extension.
+
+
+@subsection axprogramexample A Program Example
+@par
+Before getting into the technical depths of the language, it's a good idea to
+observe a small but complete AX program which could be compiled and executed.
+To begin with, here is a very simple example of a program that reads and writes
+to a particular attribute:
+@par
+@code{.c}
+// read a floating point attribute from the value of "myattribute"
+float temp = float@myattribute;
+// if the value is less than 0, clamp it to 0
+if (temp < 0.0f) float@myattribute = 0.0f;
+@endcode
+Note that there is no other required logic here to make this a compatible
+program i.e. feeding this code to the @ref vdbaxbinary "vdb_ax command line binary"
+will compile and execute over provided OpenVDB files.
+@par
+The following explains the above example in relation to OpenVDB data; OpenVDB
+points and OpenVDB volumes. The example demonstrates reading from a value on
```

Comment:
I've left this as it is, specifically because the idea of "reading" in this instance pertains to the language as well as the vdb type and I want to make the important distinction between reading and writing (where as "access" is slightly more ambiguous)